objects/src/accounts/storage/header.rs

```diff
@@ -1,14 +1,41 @@
 use alloc::vec::Vec;
 
-use vm_core::utils::{ByteReader, ByteWriter, Deserializable, Serializable};
+use vm_core::{
+    utils::{ByteReader, ByteWriter, Deserializable, Serializable},
+    ZERO,
+};
 use vm_processor::DeserializationError;
 
-use super::{AccountStorage, StorageSlotType, Word};
+use super::{
+    AccountStorage, Felt, StorageSlot, StorageSlotType, Word, NUM_ELEMENTS_PER_STORAGE_SLOT,
+};
 use crate::AccountError;
 
 // ACCOUNT STORAGE HEADER
 // ================================================================================================
 
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct StorageSlotHeader(StorageSlotType, Word);
+
+impl From<&StorageSlot> for StorageSlotHeader {
+    fn from(value: &StorageSlot) -> Self {
+        Self(value.slot_type(), value.value())
+    }
+}
+
+impl StorageSlotHeader {
+    pub fn new(value: &(StorageSlotType, Word)) -> Self {
+        Self(value.0.clone(), value.1)
```

Comment:
Let's change `StorageSlotType` to be `Copy`. This way, we won't need `clone()` here.